modules/javafx.media/src/main/native/gstreamer/plugins/dshowwrapper/dshowwrapper.cpp

```diff
@@ -1931,9 +1939,66 @@ static void dshowwrapper_load_decoder_h264(GstDShowWrapper *decoder, JFX_CODEC_I
     }
 }
 
+static gboolean dshowwrapper_reload_decoder_h264_fragmented(GstDShowWrapper *decoder, sInputFormat *pInputFormat, sOutputFormat *pOutputFormat)
+{
+    HRESULT hr = S_OK;
+    OAFilterState fs = 0;
+    IPin *pOutput = NULL;
+
+    if (decoder->pMediaControl)
+    {
+        decoder->pMediaControl->Stop();
+        decoder->pMediaControl->GetState(5000, &fs);
+    }
+
+    if (decoder->pDecoder != NULL && decoder->pGraph != NULL)
```

Comment:
Can `decoder->pDecoder` be non-null and `decoder->pGraph` be null? If so, then the decoder wouldn't be released. Does it need to be?

Reply:
Unlikely. I changed it to check for `if (decoder->pGraph == NULL)` at beginning of function and return `FALSE` if it happens, since we cannot swap decoder if graph is not initialized. Not sure if it can happen, so this check is more for just in case. Yes, we need to release decoder and re-create it during format change. I tried all possible ways to change format dynamically and nothing worked. Only complete recreation of decoder instance worked.